### ISSUE TITLE
Not finding my android device

### ISSUE THREAD

**Issue by ranit-geek** [appium/appium-docker-android#82]
hi,
please take a look at my docker compose file:

`
  appium-android:
    image: appium/appium
    privileged: true
    container_name: appium
    volumes:
      - /Volumes:/dev/bus/usb
      - ./app/1media.apk:/app/1media.apk
      - ~/.android:/root/.android
    ports:
      - 4723:4723

  java-maven:
    image: iamranit/appiumdockertest:3.0.0
    container_name: appiumdockertest
    volumes:
      - /Users/ranit/Downloads/Result:/test-output/

    command: ptest -p config.ini -t test-ui
    links:
    - appium-android:localhost

`

with this when I am running the docker compose file , everything is working properly except its not finding my android device. Error wile running:

`Starting appium ... done
Starting appiumdockertest ... done
Attaching to appium, appiumdockertest
appium            | [Appium] Appium support for versions of node < 8 has been deprecated and will be removed in a future version. Please upgrade!
appium            | [Appium] Welcome to Appium v1.8.1
appium            | [Appium] Non-default server args:
appium            | [Appium]   log: /var/log/appium.log
appium            | [Appium] Appium REST http interface listener started on 0.0.0.0:4723
appium            | [HTTP] --> POST /wd/hub/session
appium            | [HTTP] {"capabilities":{"alwaysMatch":{"appium:deviceName":"Android","appium:app":"/app/1media.apk","appium:autoGrantPermissions":"true","platformName":"Android","appium:automationName":"uiautomator2"},"firstMatch":[{}]},"desiredCapabilities":{"deviceName":"Android","app":"/app/1media.apk","platformName":"Android","automationName":"uiautomator2","autoGrantPermissions":"true"}}
appium            | [debug] [MJSONWP] Calling AppiumDriver.createSession() with args: [{"deviceName":"Android","app":"/app/1media.apk","platformName":"Android","automationName":"uiautomator2","autoGrantPermissions":"true"},null,{"alwaysMatch":{"appium:deviceName":"Android","appium:app":"/app/1media.apk","appium:autoGrantPermissions":"true","platformName":"Android","appium:automationName":"uiautomator2"},"firstMatch":[{}]}]
appium            | [debug] [BaseDriver] Event 'newSessionRequested' logged at 1534157095987 (03:44:55 GMT-0700 (PDT))
appium            | [Appium] Creating new AndroidUiautomator2Driver (v1.12.0) session
appium            | [Appium] Capabilities:
appium            | [Appium]   platformName: Android
appium            | [Appium]   deviceName: Android
appium            | [Appium]   app: /app/1media.apk
appium            | [Appium]   autoGrantPermissions: true
appium            | [Appium]   automationName: uiautomator2
appium            | [debug] [BaseDriver] W3C capabilities {"alwaysMatch":{"platformNa... and MJSONWP desired capabilities [object Object] were provided
appium            | [debug] [BaseDriver] Creating session with W3C capabilities: {"alwaysMatch":{"platformNa...
appium            | [BaseDriver] Boolean capability passed in as string. Functionality may be compromised.
appium            | [BaseDriver] Capability 'autoGrantPermissions' changed from string to boolean. This may cause unexpected behavior
appium            | [BaseDriver] Session created with session id: dd90abaa-e9ba-4411-9e54-a33b9aaa2ac9
appium            | [BaseDriver] Using local app '/app/1media.apk'
appium            | [debug] [UiAutomator2] Checking whether app is actually present
appium            | [debug] [AndroidDriver] Getting Java version
appium            | [AndroidDriver] Java version is: 1.8.0_171
appium            | [ADB] Checking whether adb is present
appium            | [ADB] Found 1 'build-tools' folders under '/root' (newest first):
appium            | [ADB]     /root/build-tools/28.0.0
appium            | [ADB] Using adb from /root/platform-tools/adb
appium            | [AndroidDriver] Retrieving device list
appium            | [debug] [ADB] Trying to find a connected android device
appium            | [debug] [ADB] Getting connected devices...
appium            | [debug] [ADB] 0 device(s) connected
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Getting connected devices...
appium            | [debug] [ADB] 0 device(s) connected
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Getting connected devices...
appium            | [debug] [ADB] 0 device(s) connected
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Getting connected devices...
appium            | [debug] [ADB] 0 device(s) connected
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Getting connected devices...
appium            | [debug] [ADB] 0 device(s) connected
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Getting connected devices...
appium            | [debug] [ADB] 0 device(s) connected
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Getting connected devices...
appium            | [debug] [ADB] 0 device(s) connected
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
appium            | [debug] [ADB] Killing adb server on port 5037
appium            | [debug] [ADB] Could not find devices, restarting adb server...
appium            | [debug] [ADB] Restarting adb
`

any help?

**Comment by butomo1989** [comment]
are you using mac? if yes, you need to ssh it first (all steps can be found in readme)

**Comment by ranit-geek** [comment]
I was following the steps in readme. But in 6th step when i am doing "adb kill-server" it is showing "adb command not found" . That is obvious because the virtual machine do not have adb installed. Can you help me here?

**Comment by butomo1989** [comment]
what it meant is to kill adb-server on your Mac machine not in VM. the description is not written well. could you create a PR for that? @ranit-geek

**Comment by ranit-geek** [comment]
I will do that. Can you tell me that where I need to execute the "docker-compose up" command ?
in my mac terminal? or in docker machine that I have created? 
the docker machine that i have do not have that docker compose file.Shall I copy the docker compose file to the docker machine and try to run the command from docker machine?

**Comment by butomo1989** [comment]
yeah, docker-compose should be run under docker-machine @ranit-geek
